### PR TITLE
Flag for camera info publishing.

### PR DESCRIPTION
**Pull request by v-mehta**
Added flag to enable/disable camera info publishing and required publisher for image transport.

This is needed so we can use other sources of camera calibration parameters like Kalibr.

### CODE REVIEW

File: pointgrey_camera_driver/src/nodelet.cpp [hunk 1]
```diff
@@ -287,7 +289,7 @@ class PointGreyCameraNodelet: public nodelet::Nodelet
     it_.reset(new image_transport::ImageTransport(nh));
     image_transport::SubscriberStatusCallback cb = boost::bind(&PointGreyCameraNodelet::connectCb, this);
     it_pub_ = it_->advertiseCamera("image_raw", 5, cb, cb);
```
Comment (servos) [inline pointgrey_camera_driver/src/nodelet.cpp]: should probably only construct one of these publishers based on the param.

**Review by servos — CHANGES_REQUESTED** [review]

File: pointgrey_camera_driver/src/nodelet.cpp (hunk 1 above)
Comment (v-mehta) [reply pointgrey_camera_driver/src/nodelet.cpp]: Good point! Will update and ping again for review.

File: pointgrey_camera_driver/src/nodelet.cpp [hunk 2]
```diff
@@ -266,9 +266,11 @@ class PointGreyCameraNodelet: public nodelet::Nodelet
     // Get the location of our camera config yaml
     std::string camera_info_url;
     pnh.param<std::string>("camera_info_url", camera_info_url, "");
+    pnh.param<bool>("publish_camera_info", publish_camera_info_, "true");
     // Get the desired frame_id, set to 'camera' if not found
     pnh.param<std::string>("frame_id", frame_id_, "camera");
 
+
```
Comment (efernandez) [inline pointgrey_camera_driver/src/nodelet.cpp:274]: remove empty line

File: pointgrey_camera_driver/src/nodelet.cpp [hunk 3]
```diff
@@ -514,10 +516,21 @@ class PointGreyCameraNodelet: public nodelet::Nodelet
             pub_->publish(wfov_image);
 
             // Publish the message using standard image transport
-            if(it_pub_.getNumSubscribers() > 0)
+            if(publish_camera_info_)
```
Comment (efernandez) [inline pointgrey_camera_driver/src/nodelet.cpp:528]: It'd be cleaner to have a functor or function that you call like this: `publish(wfov_image->image);`

That function can be set on the constructor depending on the `publish_camera_info_` param.

**Review by efernandez — APPROVED** [review]
Minor comments only.

**Review by servos — APPROVED** [review]

**Review by efernandez — APPROVED** [review]
Only a minor change remains, but consider if optional if you want.

**Comment by efernandez** [comment]
@v-mehta merge?

---
Side note: @asaini89 I can list other users here, I think the problem you're seeing are b/c new forked repos don't have contributions from us.

**Pull request merged**